p11-kit/rpc-client.c

```diff
@@ -595,6 +596,10 @@ proto_read_sesssion_info (p11_rpc_message *msg,
 	if (!p11_rpc_message_write_byte_buffer (&_msg, arr ? (*len > 0 ? *len : (uint32_t)-1) : 0)) \
 		{ _ret = CKR_HOST_MEMORY; goto _cleanup; }
 
+#define IN_BYTE_BUFFER_NULL(arr, len) \
+	if (!p11_rpc_message_write_byte_buffer_null (&_msg, len)) \
```

Comment:
These macros (`IN_BYTE_BUFFER_NULL` and `OUT_BYTE_ARRAY_NULL`) and the support function (`p11_rpc_message_write_byte_buffer_null`) are only used to handle the special case in [C_SignMessageNext](https://docs.oasis-open.org/pkcs11/pkcs11-spec/v3.1/pkcs11-spec-v3.1.pdf#page=145), which I would rather expand at the caller.